src/TableFunctions/TableFunctionOneHotEncodingView.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <TableFunctions/ITableFunction.h>
+#include <Parsers/ASTCreateQuery.h>
+#include <common/types.h>
+
+namespace DB
+{
+
+/* one_hot_encoding_view((query),column[,column,...])
+ * Turns a subquery into a table with the selected columns
+ * encoded using one hot encoding and presented
+ * as an Array(UInt8) encoded column.
+ * Used for machine learning data preparation.
+ */
+class TableFunctionOneHotEncodingView : public ITableFunction
+{
+public:
+    static constexpr auto name = "one_hot_encoding_view";
+    std::string getName() const override { return name; }
+private:
+    std::tuple<const String, const String> getWithAndSelectForColumn(String base_query_str, const String & column_name);
```

Comment:
Static?

Reply:
Yes. I will make it static.

Reply:
With the latest changes this method uses object attributes so it does not have to be static now.